packages/block-library/src/image/edit.native.js

```diff
@@ -261,6 +294,56 @@ class ImageEdit extends React.Component {
 			</BottomSheet>
 		);
 
+		//Used for display of Inspector Controls
+		const getSizeSlugDisplay = ( sizeSlugValue ) => {
+			if ( sizeSlugValue === undefined ) {
+				sizeSlugValue = DEFAULT_SIZE_SLUG;
+			}
+			const sizeDisplayItems = getSizeOptionsItems();
+			switch ( sizeSlugValue ) {
+				case IMAGE_SIZE_THUMBNAIL:
+					return sizeDisplayItems[ 0 ].label;
+				case IMAGE_SIZE_MEDIUM:
+					return sizeDisplayItems[ 1 ].label;
+				case IMAGE_SIZE_FULL_SIZE:
+					return sizeDisplayItems[ 3 ].label;
+				case IMAGE_SIZE_LARGE:
+				default:
+					return sizeDisplayItems[ 2 ].label;
+			}
+		};
+
+		const getSizeOptionsItems = () => {
+			return [
+				{ value: IMAGE_SIZE_THUMBNAIL, label: __( 'Thumbnail' ) },
+				{ value: IMAGE_SIZE_MEDIUM, label: __( 'Medium' ) },
+				{ value: IMAGE_SIZE_LARGE, label: __( 'Large' ) },
+				{ value: IMAGE_SIZE_FULL_SIZE, label: __( 'Full Size' ) },
+			];
+		};
+
+		let picker;
+
+		const getSizeOptions = () => (
```

Comment:
This ImageEdit component is getting pretty large, do you think it's worth moving the size picker to its own component? We could move the `IMAGE_SIZE_*` consts there as well, what do you think?

Reply:
I agree, I'll try for a simple refactor commit now that moves the size picker to its own component, and addresses some of the other good ideas around code you had in other comments. I'll keep the __DEV__ flag in mind as well in case minor changes will help make incorporating that nice and clean.

Reply:
Refactored to new ImageSizePicker component here: 7983f04c71df03900122d95e885749c6eb1665c2